python/pyspark/ml/wrapper.py

```diff
@@ -118,11 +118,13 @@ def _transfer_params_to_java(self):
         """
         Transforms the embedded params to the companion Java object.
         """
-        paramMap = self.extractParamMap()
         for param in self.params:
-            if param in paramMap:
-                pair = self._make_java_param_pair(param, paramMap[param])
+            if param in self._paramMap:
+                pair = self._make_java_param_pair(param, self._paramMap[param])
                 self._java_obj.set(pair)
+            if param in self._defaultParamMap:
```

Comment:
We usually make the assumption that Python defines the same default values as Java, in Spark ML at least, but given the circumstances of the JIRA - they defined their own Model - then it's still possible for `hasDefault` or the default value to return something different that Python would.  So I'm just being overly cautious here, but it's pretty cheap to just transfer the default values anyway right?